pkg/engine/task/task_apply_test.go

```diff
@@ -189,3 +190,12 @@ type errorEnhancer struct{}
 func (k *errorEnhancer) Apply(templates map[string]string, metadata renderer.Metadata) ([]runtime.Object, error) {
 	return nil, errors.New("always error")
 }
+
+func Test_namespacedGVKResources(t *testing.T) {
+	got, err := namespacedGVKResources()
+	assert.NoError(t, err)
+
+	// just a small sanity check
+	assert.True(t, got[schema.GroupVersionKind{Group: "apps", Version: "v1", Kind: "ReplicaSet"}])
+	assert.False(t, got[schema.GroupVersionKind{Group: "apiextensions.k8s.io", Version: "v1", Kind: "CustomResourceDefinition"}])
```

Comment:
can we test if a custom resource for both a Namespaced CRD and a Cluster scoped CRD also work here, just so we have it in the event of any regressions (and we're talking, in Kubernetes terms, about two different API servers here)

Reply:
You mean a CR (not CRD), right? Because CRDs are always cluster-scoped. CRs might be not.